autoload/ale/list.vim

```diff
@@ -14,11 +14,17 @@ endfunction
 function! ale#list#SetLists(buffer, loclist) abort
     if g:ale_set_quickfix
         call setqflist(a:loclist)
+        if has('patch-7.4.2200')
+            call setqflist([], 'r', {'title': expand('%')})
```

Comment:
You should include the title option in the call above, instead of making two calls.

`expand('%')` won't work all the time, as the buffer errors are being set for can be different from the current buffer. You should get the buffer's filename with `expand('#' . a:buffer)`. You may want to add either `':p'` or `':t'` to get the absolute path or the basename.